### PR TITLE
move parseMethodName func to internal/grpcutil

### PR DESCRIPTION
**Pull request by zouyee**
move parseMethodName func to internal/grpcutil

Signed-off-by: Zou Nengren <zouyee1989@gmail.com>

### CODE REVIEW

File: internal/grpcutil/util.go [hunk 1]
```diff
@@ -16,18 +16,17 @@
...
-package binarylog
+package grpcutil
 
 import (
 	"errors"
 	"strings"
 )
 
-// parseMethodName splits service and method from the input. It expects format
+// ParseMethodName splits service and method from the input. It expects format
```
Comment (easwars) [inline internal/grpcutil/util.go:26]: Couple of nits here:
The grpcutil package now has a file named `target.go` which exports a function `ParseTarget`. So, maybe to keep things consistent, we should name this file as `method.go` and the function `ParseMethod`.

File: internal/grpcutil/util_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+package grpcutil
+
+import (
+	"testing"
+)
+
+func TestParseMethodName(t *testing.T) {
+	testCases := []struct {
+		input         string
```
Comment (easwars) [inline internal/grpcutil/util_test.go:27]: s/input/methodName/
s/outputService/wantService
s/outputMethod/wantMethod
s/hasError/wantError

File: internal/grpcutil/util_test.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			if tc.hasError {
+				t.Errorf("Parsing %q got nil error, want non-nil error", tc.input)
+			}
+		}
+		if s != tc.outputService || m != tc.outputMethod {
+			t.Errorf("Parseing %q got service %q, method %q, want service %q, method %q",
+				tc.input, s, m, tc.outputService, tc.outputMethod,
+			)
+		}
+	}
```
Comment (easwars) [inline internal/grpcutil/util_test.go:57]: I think this whole thing can be simplified as follows:
```golang
if (err != nil) != tc.wantErr || s != tc.wantService || m != tc.wantMethod {
  t.Errorf("ParseMethodName(%s) = (%s, %s, %v), want (%s, %s, %v)", tc.methodName, s, m, err, tc.wantService, tc.wantMethod, tc.wantErr)
}
```

File: internal/grpcutil/util_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+func TestParseMethodName(t *testing.T) {
+	testCases := []struct {
+		input         string
+		outputService string
+		outputMethod  string
+		hasError      bool
+	}{
+		{input: "/s/m", outputService: "s", outputMethod: "m", hasError: false},
+		{input: "/p.s/m", outputService: "p.s", outputMethod: "m", hasError: false},
+		{input: "/p/s/m", outputService: "p/s", outputMethod: "m", hasError: false},
```
[the remark this answers is not in the document]
Comment (menghanl) [reply internal/grpcutil/util_test.go:34]: This is valid. It's OK to have "/" in the service part. Proto never does it, but other IDL may do it.

File: internal/grpcutil/method_test.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+		s, m, err := ParseMethod(tc.methodName)
+		if (err != nil) != tc.wantError || s != tc.wantService || m != tc.wantMethod {
+			t.Errorf("ParseMethod(%s) = (%s, %s, %v), want (%s, %s, %v)", tc.methodName, s, m, err, tc.wantService, tc.wantMethod, tc.wantError)
+		}
+		if s != tc.wantService || m != tc.wantMethod {
+			t.Errorf("Parseing %q got service %q, method %q, want service %q, method %q",
+				tc.methodName, s, m, tc.wantService, tc.wantMethod,
+			)
+		}
+	}
```
Comment (easwars) [inline internal/grpcutil/method_test.go:50]: This block is not required now, since the above check covers everything, right? Or am I missing something?

**Comment by easwars** [comment]
@zouyee Thank you for the fix !!

**Pull request merged**